File: Lab-2/e_approx.cpp
#include <iostream>
#include <string>
#include <iomanip>

using namespace std;

int factorial(int i){
    if (i==0) return 1;
    else return i * factorial(i-1);
}

double e_approximation(int i){
    if (i==0) 
    return 0;
    else 
    return (double) i/factorial(i) + e_approximation(i-1);
}


int main(){

    int i = 20;
    cout <<setprecision(11) << e_approximation(i) <<endl;

}

File: Lab-2/e_approx.py
from math import factorial
from easyinput import read_line
'''
function e_approximation(int i):
    if i==1 then
        return 0
    else:
        return i / factorial(i) + e_approx(i-1)
'''



def e_approx(i):
    if not i: return 0
    else:
        return i/factorial(i) + e_approx(i-1)

num = read_line()
while num  != "":
    num = int(num)
    res = "{:.10f}".format(e_approx(num))
    print(f"With {num} term(s) we get {res}.")
    num = int(read_line())